cli/functionary/package.py

```diff
@@ -82,6 +82,7 @@ def publish(ctx, path):
     upload_file = open(tarfile_name, "rb")
     click.echo(f"Publishing {str(tarfile_name)} package to {host}")
 
-    post("publish", files={"package_contents": upload_file})
-
-    click.echo("Publish succeded")
+    response = post("publish", files={"package_contents": upload_file})
+    upload_file.close()
```

Comment:
Look at using a context manager here as suggested earlier.  The reason why that's preferable here is that the file will still get closed even if your `post` call raises an exception.  From [the docs](https://docs.python.org/3/library/io.html#io.IOBase):

> IOBase is also a context manager and therefore supports the with statement. In this example, file is closed after the with statement’s suite is finished—even if an exception occurs:

As written here, your `close()` call would never be reached in the case of an exception.